extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -0,0 +1,347 @@
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {Layout, isLayoutSizeDefined} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {VideoEvents} from '../../../src/video-interface';
+
+import {assertAbsoluteHttpOrHttpsUrl, tryDecodeUriComponent} from '../../../src/url';
+import {dev, user} from '../../../src/log';
+import {fullscreenEnter, fullscreenExit, isFullscreenElement, removeElement} from '../../../src/dom';
+import {getData, listen} from '../../../src/event-helper';
+import {getIframe} from '../../../src/3p-frame';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+
+/**
+ * @implements {../../../src/video-interface.VideoInterface}
+ */
+class AmpViqeoPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?string} */
+    this.videoId_ = null;
+
+    /** @private {?string} */
+    this.profileId_ = null;
+
+    /** @private {?string} */
+    this.markTagParams_ = null;
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?number} */
+    this.volume_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?Object} */
+    this.viqeoPlayer_ = null;
+
+    /** @private {boolean} */
+    this.kindIsProd_ = true;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://static.viqeo.tv', opt_onLayout);
+    this.preconnect.url('https://stage.embed.viqeo.tv', opt_onLayout);
+  }
+
+  /**
+   * @param {!Layout} layout
+   * @return {boolean}
+   * @override
+   */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    return null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.videoId_ = user().assert(
+        this.element.getAttribute('data-videoid'),
+        'The data-videoid attribute is required for <amp-viqeo-player> %s',
+        this.element);
+
+    this.profileId_ = user().assert(
+        this.element.getAttribute('data-profileid'),
+        'The data-profileid attribute is required for <amp-viqeo-player> %s',
+        this.element);
+
+    this.markTagParams_ = this.element.getAttribute('data-tag-settings');
+
+    this.kindIsProd_ = this.element.getAttribute('data-kind') !== 'stage';
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+
+  }
+
+  /** @override */
+  layoutCallback() {
+    let scriptPlayerInit = this.element.getAttribute('data-script-url');
```

Comment:
you have access to all `data-` params inside the `3p` script. Please move this logic there, there is no need manipulate and send these through another json object.